### PR TITLE
Update `processcreds.CredentialProcessResponse` visibility to public

### PR DESCRIPTION
**Pull request by joshdk**
In order to have a canonical type for encoding/decoding JSON for a credential process, the visibility of this type needed to be public. This type has been detailed in official AWS CLI documentation and is effectively part of the tooling ecosystem's external interface.

### CODE REVIEW

File: credentials/processcreds/provider.go
Comment (joshdk) [inline credentials/processcreds/provider.go:155]: Comment taken from AWS CLI `credential_process` authoring documentation.

See https://docs.aws.amazon.com/sdkref/latest/guide/feature-process-credentials.html#feature-process-credentials-output.

File: credentials/processcreds/provider.go
Comment (joshdk) [inline credentials/processcreds/provider.go:159]: Comments copied from the `types.Credentials` struct fields.

See https://github.com/aws/aws-sdk-go-v2/blob/main/service/sts/types/types.go#L34.

**Review by joshdk — COMMENTED** [review]
For context - I am authoring a custom tool for use as a credential_process, which requires a type that I can instantiate and json.Marshal to stdout.

I am currently duplicating this type, but referencing a canonical type from the package (which is also handling AWS profile manipulation) would be the semantically correct thing, and would avoid inadvertent tool breakages when e.g. upgrading the SDK version.

For consistency, and identical change was made in https://github.com/aws/aws-sdk-go/pull/4621.

Thank you for your consideration!

**Comment by joshdk** [comment]
Friendly 1-week ping if anyone would like to approve the CI workflows 🙂

**Comment by joshdk** [comment]
Friendly 1-month ping for some engagement 🙂

**Review by aajtodd — APPROVED** [review]

**Comment by aajtodd** [comment]
@joshdk 

Thanks for the PR and sorry for the delay. I think I agree that this is effectively part of the overall ecosystem tooling interface. Making this type public should be ok. It looks like CI checks didn't run yet, can you try and push up an empty commit to see if it triggers an approval? Thanks.

**Comment by joshdk** [comment]
Thank you so much @aajtodd!

I rebased off of the latest `main` and it appears that the CI checks are now waiting for a maintainer to approve.

**Pull request merged**
